src/extensions/agent_modules/shared/state_store/src/agent_state_store.c

```diff
@@ -496,10 +188,12 @@ int ADUC_StateStore_GetDeviceRegistrationStatePollIntervalSeconds()
  * @brief Get the Device Update service device ID.
  * @return The device ID, or NULL if not found.
  */
-const char* ADUC_StateStore_GetDeviceId(void)
+const char* ADUC_StateStore_GetDeviceId()
```

Comment:
All of these getters should return copies of the value instead of the actual value or the callers need to lock the semaphore and manage access themselves.

Reply:
good catch. the pointers need an alloc and deep copy in general. Fixed with this commit:[make copies of strings for state store getters](https://github.com/Azure/iot-hub-device-update/pull/584/commits/8635ed011ebfc7396ecc82ffe15532439285e6df)